api/s2n.h

```diff
@@ -425,6 +426,56 @@ extern int s2n_connection_is_ocsp_stapled(struct s2n_connection *conn);
 S2N_API
 extern struct s2n_cert_chain_and_key *s2n_connection_get_selected_cert(struct s2n_connection *conn);
 
+/**
+ * Returns the length of the s2n certificate chain `chain_and_key`.
+ * 
+ * @param chain_and_key A pointer to the s2n_cert_chain_and_key object being read.
+ * @param cert_length This return value represents the length of the s2n certificate chain `chain_and_key`.
+ */
+S2N_API
+extern int s2n_get_cert_chain_length(struct s2n_cert_chain_and_key *chain_and_key, uint32_t *cert_length);
+
+/**
+ * Returns the certificate `out_cert` present at the index `cert_idx` of the certificate chain `chain_and_key`.
+ * 
+ * Note that the index of the `head_cert` is zero. If the certificate chain `chain_and_key` is NULL or the
```

Comment:
I'm not sure I follow the reference to head_cert? Is that the root or leaf cert? Maybe use root/leaf terminology instead?